src/pickers/demo.html

```diff
@@ -0,0 +1,194 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0"/>
+  <title>Date picker</title>
+  <style type="text/css">
+    body {
+      background-color: #eee;
+    }
+
+    body > div {
+      padding-top: 0 !important;
+    }
+  </style>
+</head>
+<body>
+
+  <div class="mdl-layout mdl-js-layout mdl-layout--fixed-drawer">
+    <header class="mdl-layout__header">
+        <div class="mdl-layout__header-row">
+        <!-- Title -->
+        <span class="mdl-layout-title">Date picker demo</span>
+        <!-- Add spacer, to align navigation to the right -->
+        <div class="mdl-layout-spacer"></div>
+      </div>
+    </header>
+    <div class="mdl-layout__drawer">
+      <span class="mdl-layout-title">Date picker demo</span>
+      <nav class="mdl-navigation">
+        <a class="mdl-navigation__link" href="#attached">Attached to input</a>
+        <a class="mdl-navigation__link" href="#fixed">Fixed picker</a>
+        <a class="mdl-navigation__link" href="#custom-format">Custom format</a>
+        <a class="mdl-navigation__link" href="#custom-range">Custom range</a>
+        <a class="mdl-navigation__link" href="#inline">Inline picker</a>
+        <a class="mdl-navigation__link" href="#landscape">Landscape picker</a>
+      </nav>
+    </div>
+    <main class="mdl-layout__content">
+      <div class="page-content">        
+        <div class="mdl-grid">
+          <div class="mdl-cell">
+            <div><a name="attached"></a></div>
+            <h6>Datepicker attached to input</h6>
+            
+            <div id="demo-picker" class="mdl-textfield mdl-js-textfield mdl-textfield--floating-label mdl-datepicker mdl-js-datepicker">
+              <input type="text" class="mdl-textfield__input mdl-datepicker__input" required>
```

Comment:
I'm curious... what about if ```type="date"```? By default the ```input``` will render its default format _"mm/dd/yyyy"_ and the final result, I think, will be [like this](http://puu.sh/oY6kB/18c13266a2.png)...